src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java

```diff
@@ -184,13 +184,16 @@ private void writeFileClosingTag() {
     public void addError(AuditEvent event) {
         if (event.getSeverityLevel() != SeverityLevel.IGNORE) {
             final String fileName = event.getFileName();
-            if (fileName == null || !fileMessages.containsKey(fileName)) {
-                writeFileError(event);
-            }
-            else {
+            final boolean fileNameExists = fileName != null
+                    && fileMessages.containsKey(fileName);
+
+            if (fileNameExists) {
```

Comment:
```suggestion
            if (fileMessages.containsKey(fileName)) {
```
good idea to make positive code flow.